sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_internal/polling.py

```diff
@@ -11,3 +13,22 @@ def __init__(self):
 
     def get_final_get_url(self, pipeline_response):
         return None
+
+    def set_initial_status(self, pipeline_response):
+        self._request = pipeline_response.http_response.request
+        response = pipeline_response.http_response
+
+        self._set_async_url_if_present(response)
+
+        if response.status_code in {200, 201, 202, 204} and self._async_url:
+            try:
+                return self.get_status(pipeline_response)
+            except BadResponse:
+                return "InProgress"
```

Comment:
I think this method could reasonably assume the response is from one of the expected APIs. Do we really want to carry on when it isn't? With that assumption and an override for `get_polling_url`, you also don't need to call `_set_async_url_if_present` or read `_async_url`.

Reply:
That sounds reasonable. It makes sense to avoid the call to `_set_async_url_if_present`, since we know it is present. I don't know if an override for `get_polling_url` is necessary though -- while `_async_url` is only used there if we remove the read here, it makes sense to me to still set the attribute since the superclass is expecting it

Reply:
It's paranoia making me write this, but a future version of azure-core that renames `_async_url` or changes the way `OperationResourcePolling.get_polling_url` uses it may break this code. I'm comfortable relying on the stability of public method behavior, but in this case we're relying on the private state determining that behavior.

Reply:
Ah, I see -- that completely makes sense. I'll make that change